akka-http-core/src/main/java/akka/http/javadsl/model/ContentRange.java

```diff
@@ -30,7 +31,7 @@ public static ContentRange create(long first, long last, long instanceLength) {
     }
     @SuppressWarnings("unchecked")
     public static ContentRange create(long first, long last, OptionalLong instanceLength) {
-        return ContentRange$.MODULE$.apply(first, last, OptionConverters.toScala(instanceLength));
+        return ContentRange$.MODULE$.apply(first, last, OptionConverters.toScala(instanceLength).map(length -> (Object)length));
```

Comment:
`OptionConverters.toScala(OptionalLong)` returns `Option[java.lang.Long]` so this is just Java not understanding variance (as expected), however, it is surprising that it thinks that `Option[Long]` in Scala is `Option<Object>`.

Actually mapping involves allocation and a lambda that will actually be invoked, we don't want that, just force-cast is probably ok `(Option<Object>)(Object) OptionConverters.toScala(instanceLength)` even if it feels a bit dirty.

Reply:
Or if we want to keep it slightly cleaner, this should compile fine `(instanceLength.isPresent() ? Option.apply(instanceLength.getAsLong()) : Option.empty())`